### PR TITLE
add new use cases

### PR DESCRIPTION
**Pull request by shardhrv**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -391,6 +391,210 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+**Preconditions**: User is in search mode.
+
+**MSS**
+1. User enters search mode.
+2. User specifies a contact to exclude from search results.
+3. PlanPal excludes the contact from future searches and displays a success message.
+
+   Use case ends.
+
+**Extensions**
```
Comment (jan-kai1) [inline docs/DeveloperGuide.md:558]: Extension 2a seems to follow path of UC13, maybe could remove to focus UC14 on the clearing of exclusion or inclusion path to UC 13 instead

**Review by jan-kai1 — CHANGES_REQUESTED** [review]

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -391,6 +391,212 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+* 2a. User enters invalid information (e.g., telegram name longer than 32 characters).
+    * 2a1. PlanPal displays an error message.
+    * 2a2. User re-enters valid information.
+
+      Use case resumes from step 3.
+
+**UC09: Remove a contact from an event**
+
+**MSS**
+1. User specifies the event they want to modify.
```
Comment (chuajunyu) [inline docs/DeveloperGuide.md:468]: Add event not found extension here too?

**Review by chuajunyu — CHANGES_REQUESTED** [review]

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -391,6 +391,212 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+   Use case ends.
+
+**Extensions**
+* 2a. User specifies an invalid or non-existent contact.
+    * 2a1. PlanPal displays an error message indicating the contact is invalid.
+    * 2a2. User re-enters a valid contact.
+
+      Use case resumes from step 3.
+
+**UC14: Include an excluded contacts in search mode**
```
Comment (jan-kai1) [inline docs/DeveloperGuide.md:548]: I think change this to clear all excluded, we do not have a include command currently
```suggestion
**UC14: Clear excluded contacts in search mode**

**Preconditions**: User is in search mode.

**MSS**
1. User specifies command to clear currently excluded contacts
2. PlanPal clears the currently excluded contacts and displays a success message.
3. Excluded contacts are added back to the search list.

    Use case ends
**Extensions**

* 1a. User has no currently exclude contacts
    *1a1: PlanPal displays a message that user has no currently excluded contacts
    
    Use case ends
        


```

**Review by jan-kai1 — CHANGES_REQUESTED** [review]

**Review by chuajunyu — APPROVED** [review]
LGTM

**Pull request merged**